docs/src/building.md

```diff
@@ -344,3 +344,40 @@ Examples of builders that depend on other binaries include:
 * [`Xorg_libX11`](https://github.com/JuliaPackaging/Yggdrasil/blob/eb3728a2303c98519338fe0be370ef299b807e19/X/Xorg_libX11/build_tarballs.jl#L36-L42)
   depends on `Xorg_libxcb_jll`, and `Xorg_xtrans_jll` at build- and run-time,
   and on `Xorg_xorgproto_jll` and `Xorg_util_macros_jll` only at build-time.
+
+# Building and testing JLL packages locally
+
+As a package developer, you may want to test JLL packages locally, or as a binary dependency
+developer you may want to easily use custom binaries.  In both cases, the new `dev` and
+`overrides` directory support make it easy to get complete control over the local JLL package
+state.
+
+## Overriding a prebuilt JLL package's binaries
+
+After running `pkg> dev LibFoo_jll`, a local JLL package will be checked out to your depot's
+`dev` directory, (on most installations this is `~/.julia/dev`) and by default it will make
+use of binaries within your depot's `artifacts` directory.  However, if an `override`
+directory is present within that package directory, it will look within that directory for
+binaries, rather than in any artifact directory.  There is no mixing and matching of binaries
+within a single JLL package; if an `override` directory is present, all products defined within
+that JLL package must be found within the `override` directory, none will be sourced from an
+artifact.  Dependencies (e.g. found within another JLL package) may still be loaded from their
+respective artifacts, so dependency JLLs must themselves be `dev`'ed and have `override`
+directories created with files or symlinks created within them.
+
+### Auto-populating `override` directory
+
+To ease creation of an `override` directory, JLL packages now contain a `dev_jll()` function,
```

Comment:
I am a bit torn of the use of "now" here, for the reason I already outlined above: how will this read to somebody in one year? One way to go about this wold be to insert a date in here, e.g. "since August 2020". But that doesn't seem great, also it depends a bit on when the patch is merged and Yggdrasil updated to use it? 

So I guess instead I would suggest this: drop "now" here, and instead insert a comment somewhere along the lines of: "Note that older JLLs which were generated before this feature was introduced may lack a `dev_jll`
function." 
```suggestion
To ease creation of an `override` directory, JLL packages contain a `dev_jll()` function,
```